docs/connect/clients/nodejs.md

```diff
@@ -136,166 +136,75 @@ await client.disconnect();
 
-```js
-await Promise.all([
-    client.json.set('user:1', '$', {
-        "name": "Paul John",
-        "email": "paul.john@example.com",
-        "age": 42,
-        "city": "London"
-    }),
-    client.json.set('user:2', '$', {
-        "name": "Eden Zamir",
-        "email": "eden.zamir@example.com",
-        "age": 29,
-        "city": "Tel Aviv"
-    }),
-    client.json.set('user:3', '$', {
-        "name": "Paul Zamir",
-        "email": "paul.zamir@example.com",
-        "age": 35,
-        "city": "Tel Aviv"
-    }),
-]);
+```typescript
+const client = createClient({
+  // ... client options
+});
+// Always ensure there's a listener for errors in the client to prevent process crashes due to unhandled errors
+client.on('error', error => {
+    console.error(`Redis client error: ${error}`)
+});
 ```
 
-Let's find user 'Paul` and filter the results by age.
 
-```js
-let result = await client.ft.search(
-    'idx:users',
-    'Paul @age:[30 40]'
-);
-console.log(JSON.stringify(result, null, 2));
-/*
-{
-  "total": 1,
-  "documents": [
-    {
-      "id": "user:3",
-      "value": {
-        "name": "Paul Zamir",
-        "email": "paul.zamir@example.com",
-        "age": 35,
-        "city": "Tel Aviv"
-      }
-    }
-  ]
-}
- */
-```
+#### Handling reconnections
 
-Return only the city field.
+If the socket unexpectedly closes, such as due to network issues, the client rejects all commands already sent, as they might have been executed on the server.
+The rest of pending commands will remain queued in memory until a new socket is established.  
+The client uses `reconnectStrategy` to decide when to attempt to reconnect. 
+The default strategy is to calculate delay before each attempt based on the attempt number `Math.min(retries * 50, 500)`. You can customize this strategy by passing a supported value to `reconnectStrategy` option:
 
-```js
-result = await client.ft.search(
-    'idx:users',
-    'Paul @age:[30 40]',
-    {
-        RETURN: ['$.city']
-    }
-);
-console.log(JSON.stringify(result, null, 2));
-
-/*
-{
-  "total": 1,
-  "documents": [
-    {
-      "id": "user:3",
-      "value": {
-        "$.city": "Tel Aviv"
-      }
+1. Define a callback `(retries: number, cause: Error) => false | number | Error` **(recommended)**
+```typescript
+const client = createClient({
+  socket: {
+    reconnectStrategy: function(retries) {
+        if (retries > 20) {
+            console.log("Too many attempts to reconnect. Redis connection was terminated");
+            return new Error("Too many retries.");
+        } else {
+            return retries * 500;
+        }
     }
-  ]
-}
- */
+  }
+});
+client.on('error', error => {
+    console.error(`Redis client error: ${error}`)
+});
 ```
- 
-Count all users in the same city.
+In the provided reconnection strategy callback, the client attempts to reconnect up to 20 times with a delay of `retries * 500` milliseconds between attempts. 

```

Comment:
This is wrong..

Reply:
This sentence refers to the code example above, where we demonstrate how to throw an Error, etc.

Reply:
oh my bad, totally missed it